### PR TITLE
add default connection timeout to http-core

### PR DESCRIPTION
**Pull request by willward**
# Pull Request Template

## Description

adding a default http timeout of 60 seconds on requests in the js-sdk. The timeout is for connection to the server and first byte of response received.

Fixes # (issue)

SPARK-83465 - ETIMEDOUT and then next download hung

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

added 2 new tests to `request.js` in `http-core/test/integration/spec` : 
- "ensure default connection timeout is set"
- "ensure default connection timeout can be overwritten"

ran new and existing tests using command `npm test -- --packages @webex/http-core --node`

**Test Configuration**:
* SDK Version - v1.74.5
* Node/Browser Version - v10.15.3
* NPM Version - 6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/http-core/src/index.js [hunk 1]
```diff
@@ -40,6 +41,10 @@ const protorequest = curry(function protorequest(defaultOptions, options) {
     Reflect.deleteProperty(options, 'json');
   }
 
+  if (!options.timeout) {
+    options.timeout = 60000; // default 60 second timeout
```
Comment (mattnorris) [inline packages/node_modules/@webex/http-core/src/index.js]: Let's move this into a `config.js` in the same folder. [Encryption](https://github.com/mattnorris/spark-js-sdk/blob/3ba8c4c442b757808f9dfd4cc4b991013cb260cb/packages/node_modules/%40webex/internal-plugin-encryption/src/config.js#L17) has some good examples.

**Comment by willward** [comment]
Hi Matt,
             I might need to be extra careful on this default, after adding the ediscovery timeouts we hit an issue calling one of the APIs with latency greater than the timeout, I thought this would be ok as the timeout is for socket connection and first byte transferred. But in this case the API was taking up to 30 seconds processing and was not returning any response until all the processing was done and so it triggered the timeout. This could trip us up for other APIs if we introduce such a widespread default. Perhaps it still ok to proceed but double it up to 2 minutes.

Can you think of any approach where the consumer of the SDK could choose the default timeout at init time or something like this, so that we could enforce a default timeout in our app but not affect other consumers ?

**Comment by mattnorris** [comment]
@willward wouldn't you just leave the code as-is and pass `options.timeout`, i.e., make no changes to the code? That would give you control and not affect other consumers. 

I also just added [a limit to prevent an infinite loop](https://github.com/webex/webex-js-sdk/pull/1284) while renegotiating keys. Does this solve your original problem?

**Comment by willward** [comment]
> @willward wouldn't you just leave the code as-is and pass `options.timeout`, i.e., make no changes to the code? That would give you control and not affect other consumers.
> 
> I also just added [a limit to prevent an infinite loop](https://github.com/webex/webex-js-sdk/pull/1284) while renegotiating keys. Does this solve your original problem?

@mattnorris yes i think that will help alright, there is also one other api we use in the encryption plugin ( _fetchDownloadUrl ) which does the http request of POST files/download/endpoints which does not have a timeout set which i was trying to cover with this. Probably better if it is done specific for that API as opposed to this wide ranging default ( which i now think is much too risky ). I'll close this PR thanks for looking into it.

**Pull request closed without merging**